packages/babel-parser/test/fixtures/typescript/cast/unparenthesized-assert-and-assign/output.json

```diff
@@ -3,7 +3,8 @@
   "start":0,"end":46,"loc":{"start":{"line":1,"column":0,"index":0},"end":{"line":2,"column":29,"index":46}},
   "errors": [
     "SyntaxError: Invalid left-hand side in assignment expression. (1:0)",
-    "SyntaxError: Invalid left-hand side in assignment expression. (2:6)"
+    "SyntaxError: Invalid left-hand side in assignment expression. (2:6)",
+    "SyntaxError: Invalid left-hand side in object destructuring pattern. (2:6)"
```

Comment:
We are reporting two errors at the same position, for the same invalid syntax. We should use `raiseOverwrite` and only report the second one.

Reply:
So, this isn't actually anything new and is just following the "current behavior" or recursively giving you every left-hand-side error in an expression. You can test this by using the current babel with something like this:

```javascript
require("@babel/parser").parse("({ a: ++5 } = 10)", { errorRecovery: true }).errors
```

Which will give you an almost identical result:

```
SyntaxError: Invalid left-hand side in prefix operation. (1:8)
SyntaxError: Invalid left-hand side in object destructuring pattern. (1:6)
```

In fact, you can create a pathological case by just doing:

```javascript
require("@babel/parser").parse("5++++++", { errorRecovery: true }).errors
```

And that will look even worse, since they'll "seem" like duplicated bugs, just because they all start at the same position (but in reality would look different if we tracked error *ranges*):

```
SyntaxError: Invalid left-hand side in postfix operation. (1:0)
SyntaxError: Invalid left-hand side in postfix operation. (1:0)
SyntaxError: Invalid left-hand side in postfix operation. (1:0)
SyntaxError: Invalid left-hand side in postfix operation. (1:0)
```

Anyways, all this to say, I think if we want different behavior for nested LHS errors, we should file a separate bug for exploring the raiseOverwrite thing, but this current test is simply surfacing that we now correctly don't ignore this LHS bug and just happens to surface this existing behavior, which to fix would require exploring and unrelated thing and potentially changing other test results and behaviors.